docs/DeveloperGuide.adoc

```diff
@@ -729,55 +729,303 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
 |`* *` |user |hide link:#private-contact-detail[private contact details] by default |minimize chance of someone else seeing them by accident
 
+|`* *` |power user |launch a shell version of the product |operate without the clutter
+
+|`* *` |user |record schedules with my contacts (like a calendar) | keep track of my upcoming events
+
+|`* *` |user |see my contact’s address on google map |know which area do they resides in visually and know how far is it from my current location
+
+|`* *` |power user |exclude words in my searches |filter my search result
+
+|`* *` |user |copy contact details to clipboard | have quick access to contact details in plaintext
+
+|`* *` |user |store social media information for contacts |keep track
+
+|`* *` |user |find out the route from one contact’s address to my home address |know the way to my contact’s home
+
+|`* *` |user |synchronize contact information with all devices |choose not to update one by one
+
+|`* *` |forgetful user |input events based on natural language |add in events easily
+
+|`* *` |careless user |specify number of times to undo/redo with in input parameter |save time
+
+|`* *` |user |get a list of contact that I removed (like a Recycle bin) |recover the accidentally deleted ones
+
+|`* *` |user |remove a contact permanently from the deleted contact list (like empty Recycle bin) |keep the contact list clean
+
 |`*` |user with many persons in the address book |sort persons by name |locate a person easily
-|=======================================================================
 
-{More to be added}
+|`*` |user |send an email from the product (launch the system’s default mailto:) |do something with the contacts
+
+|`*` |user |customize the URL of my contact (instead of the default google page) |be up to date with how my contacts are doing
+
+|`*` |user |get notification on contacts’ birthdays |send birthday wishes
+
+|`*` |user |get notification on upcoming events |choose not to remember mentally
+
+|`*` |user |store standalone events (does not involve anyone in the contact list) |keep track of my schedule
+
+|`*` |user |tag the hobbies of contacts |know what to do with friends in parties
+
+|`*` |user |get contact details by scanning QR code |choose not to type manually
+
+|`*` |user |find date matches |meet new friends
+
+|`*` |user |share events online for people to see |people can join me
+|=======================================================================
 
 [appendix]
 == Use Cases
 
 (For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
 
 [discrete]
-=== Use case: Delete person
+=== Use case: Add contact by name
+
+*MSS*
+
+1.  User input command to add contact: add [NAME]
+2.  Address book launches a window prompting user to fill in required fields (phone number, address...)
```

Comment:
the project requires you to implement one shot commands. in light of that, this use-case may not be very appropriate